src/EnergyPlus/MixedAir.cc

```diff
@@ -3683,11 +3684,13 @@ namespace MixedAir {
         }
 
         // Apply Maximum Fraction of Outdoor Air Schedule
+        Real64 currentMaxOAMassFlowRate = this->MaxOAMassFlowRate;
         if (this->MaxOAflowSchPtr > 0) {
             Real64 MaxOAflowfracVal = GetCurrentScheduleValue(this->MaxOAflowSchPtr);
             MaxOAflowfracVal = min(max(MaxOAflowfracVal, 0.0), 1.0);
+            currentMaxOAMassFlowRate = min(this->MaxOAMassFlowRate, this->MixMassFlow * MaxOAflowfracVal);
             OutAirMinFrac = min(MaxOAflowfracVal, OutAirMinFrac);
-            this->OAMassFlow = min(this->OAMassFlow, this->MixMassFlow * MaxOAflowfracVal);
+            this->OAMassFlow = min(this->OAMassFlow, currentMaxOAMassFlowRate);
```

Comment:
Just talking out loud here. So if OA is limited by the max schedule then it will go to 0 if the user so chooses. This is good, but if the exhaust fan is still running then what happens?

Reply:
Then the return is forced to match the supply minus available OA (if OA is zero, then return=supply). The zone has to make up the exhaust flow with infiltration/mixing/zoneventilation.  If there's not enough of that, then the zone balance warning should trip.